go-controller/pkg/clustermanager/userdefinednetwork/controller_helper.go

```diff
@@ -76,12 +77,24 @@ func (c *Controller) updateNAD(obj client.Object, namespace string) (*netv1.Netw
 		return nil, fmt.Errorf("foreign NetworkAttachmentDefinition with the desired name already exist [%s/%s]", nadCopy.Namespace, nadCopy.Name)
 	}
 
-	if reflect.DeepEqual(nadCopy.Spec.Config, desiredNAD.Spec.Config) && reflect.DeepEqual(nadCopy.ObjectMeta.Labels, desiredNAD.ObjectMeta.Labels) {
+	// NAD update path, need to merge internal (k8s.ovn.org) current annotations with desired
+	for k, v := range nadCopy.Annotations {
+		if strings.HasPrefix(k, types.OvnK8sPrefix) {
+			if desiredNAD.Annotations == nil {
+				desiredNAD.Annotations = make(map[string]string)
+			}
+			desiredNAD.Annotations[k] = v
+		}
+	}
```

Comment:
So what we are saying is that the UDN owns all annotations on the NAD except the OVNK annotations, instead of just owning the annotations that are set on the UDN.

Lets say UDN has annotations A,B,C. Let's say NAD has annotations A,B,D. Let's say none of this annotations are OVNK annotations. The end result would be NAD with annotations A,B,C instead of NAD with annotations A,B,C,D. 

So nobody except OVNK can set and upkeep annotations on NADs owned by UDNs. 

Not seeing a specific problem here just want to make sure this is on purpose.